src/string_id_null_ids.cpp

```diff
@@ -1,34 +1,41 @@
 #include "string_id.h"
 
-// These are centralized in this file because they must appear in one and only one compilation unit,
-// and it's problematic to define them in the respective cpp files for each class.
-// Very repitious, so define them with a macro.
-#define MAKE_NULL_ID( type, ... ) \
-class type; \
-template<> const string_id<type> &string_id<type>::NULL_ID() { \
-    static string_id<type> id = string_id<type>( __VA_ARGS__ ); \
-    return id; \
-}
+#define MAKE_NULL_ID( type, ... )  class type;  template<> const string_id<type> &string_id<type>::NULL_ID() {  static string_id<type> id = string_id<type>( __VA_ARGS__ );  return id; }
```

Comment:
commented and readable -> uncommented and unreadable
What makes you think this is an ok change? :/